_posts/2018-04-09-middleware-event.md

```diff
@@ -0,0 +1,191 @@
+make sure that the passed parameter has the correct value, since objects are passed by reference, which means that if the 
+variable gets modified, the value will be updated for both parties using it. 
+
+{% highlight php %}
+
+use Symfony\Component\EventDispatcher\Event;
+
+class Foo extends Event
+{
+    private $event;
+    private $container;
+
+    public function __construct($event, DataContainer $container)
+    {
+        $this->event = $event;
+        $this->container = $container;
+    }
+
+    public function getEvent()
+    {
+        return $this->event;
+    }
+
+    public function getContainer(): DataContainer
+    {
+        return $this->container;
+    }
+}
+{% endhighlight %}
+
+### The DataContainer class
+Below is the DataContainer class, used to add some new data to the existing event. 
+
+{% highlight php %}
+
+class DataContainer
+{
+    /** @var array */
+    private $data;
+
+    public function __construct(array $data)
+    {
+        $this->data = $data;
+    }
+
+    public function addData(string $key, $value)
+    {
+        $this->data[$key] = $value;
+    }
+}
+
+{% endhighlight %}
+
+## Manipulate the event
+Now you will need something that can manipulate the existing event. Creating an event subscriber that subscribes to the 'foo'
+event is a good start. Now it's up to you to decide what your subscriber should do.     
+
+In this case imagine that we need a users age and address for later use, but they are both missing in the SimpleBus event data that
+was provided from our application. This means that we somehow want to add these values to the event, before it gets passed
+on to another application. This will be done by the decorator class, which is a simple event subscriber. 
+
+### Example of a decorator class
+This example refers to when a new user has registered to your application.
+
+{% highlight php %}
+
+class UserHasRegisteredDecorator implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return [
+            'foo' => 'decorate',
+        ];
+    }
+    /**
+     * Populates the event with more specific data based on what is needed for the specific event.
+     *
+     * @param Foo $event
+     */
+    public function decorate(Foo $event)
+    {
+        // Retrieve the DataContainer object
+        $container = $event->getContainer();
+
+        // Check if the message and assigned event match
+        if ($event->getEvent() instanceof UserHasRegistered) {
+        
+            $user = $this->getEntityManager()->getRepository(User::class)->find($event->getId()); 
+            // Add the data you want
+            $container->addData('age', $user->getAge());
+            $container->addData('address', $user->getAddress());
+        }
+    }
+}
+{% endhighlight %}
+
+Now we have successfully added age and address to the specific event data before it gets passed on to another application. 
```

Comment:
What do you mean with "to the specific event data"?

Reply:
Changed it to just "event data", using specific will just confuse the reader :)